### ISSUE TITLE
[Bug]: 5.0.4 Android login creates unsubscription after 2 seconds unless permission is granted

### ISSUE THREAD

**Issue by johanlantz** [OneSignal/OneSignal-Android-SDK#1927]
### What happened?

Just calling login with the Android sdk creates an unsubscribe event after 2 seconds unless the notification permission is granted. 

![Screenshot 2023-12-01 at 17 00 48](https://github.com/OneSignal/OneSignal-Android-SDK/assets/7461655/363edae7-13e0-4045-b85f-70d37ed4890f)

### Steps to reproduce?

```Markdown
I used the Flutter example app and pressed login.
In Android it creates a Last Unsubscribed entry after two seconds unless I quickly grant the notification permission.

Same test using the flutter app on an iPhone does not create the Last Unsubscribed event.
```

### What did you expect to happen?

Last Unsubscribed should not be created by just calling login.

### OneSignal Android SDK version

5.0.4

### Android version

13

### Specific Android models

```Markdown
Pixel 7a with Android 14
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by PachecoLpG** [comment]
Same problem here, multiple unsubscribes on different devices

**Comment by VitalyTimofeev** [comment]
The same problem with OneSignal 5.1.1 SDK version. We have following code to init sdk: 

`OneSignal.setConsentRequired(true);`
 `OneSignal.setConsentGiven(true);`
 `OneSignal.initWithContext(this, ONE_SIGNAL_APP_ID);`
 
 and following code to request permission after init:

`    fun requestPushPermission() {
        scope.launch {
            OneSignal.Notifications.requestPermission(true)
        }
    }
`
Here are our findings when testing different cases:
- if you first install the app being in Canada - your device is unsubscribed just in few seconds after. Delivery statistic shows your devices as "Unsubscribed"
- if you connect to European VPN and restart the app - your device is subscribed. Pushes arriving fine
- if you disconnect from VPN and restart the app to be in Canada again - your device still subscribed. Pushes arriving fine.
 
 This behaviour appeared after migration from OneSignal 4.8.1 to 5.1.1.

**Comment by jkasten2** [comment]
We have made some changes that should have addressed this. Can you test again on [5.1.15](https://github.com/OneSignal/OneSignal-Android-SDK/releases/tag/5.1.15)?
* Included in [Flutter 5.2.1](https://github.com/OneSignal/OneSignal-Flutter-SDK/releases/tag/5.2.1) if you are using that SDK.
